### PR TITLE
Miscellaneous Vulkan refactoring.

### PR DESCRIPTION
**Pull request by prideout**
Remove unnecessary depth attachment on the swap chain, introduce a "work" command buffer, etc.  This does not change anything major, it is just prep work.

### CODE REVIEW

File: filament/src/driver/vulkan/VulkanHandles.cpp [hunk 1]
```diff
@@ -154,17 +154,11 @@ VkExtent2D VulkanRenderTarget::getExtent() const {
...
-    return getSwapContext(mContext).attachment;
+    return mOffscreen ? mColor : getSwapContext(mContext).attachment;
 }
 
 VulkanAttachment VulkanRenderTarget::getDepth() const {
-    if (mOffscreen) {
-        return mDepth;
-    }
-    return mContext.currentSurface->depth;
+    return mOffscreen ? mDepth : VulkanAttachment {};
```
Comment (romainguy) [inline filament/src/driver/vulkan/VulkanHandles.cpp]: At some point we'll need to move this decision out of the driver. EGL has the same issue: our drivers "know" that the onscreen buffer doesn't need depth. Also, what is "mOffscreen" here? Not all of our off-screen buffers need depth (only the one used by the color pass, all the others currently don't need depth).

Comment (prideout) [reply filament/src/driver/vulkan/VulkanHandles.cpp]: Understood -- it should be easy to resurrect the depth buffer in the future when we do this.

If it is not a swap chain buffer, then mOffscreen is true, therefore VulkanRenderTarget owns the attachments, so we return mDepth.  Yes, mDepth is non-null only if the render target was created with TargetBufferFlags::DEPTH.

Comment (romainguy) [reply filament/src/driver/vulkan/VulkanHandles.cpp]: SG thanks

File: filament/src/driver/vulkan/VulkanDriverImpl.cpp [hunk 2]
```diff
@@ -203,6 +203,17 @@ void createVirtualDevice(VulkanContext& context) {
...
     vmaCreateAllocator(&allocatorInfo, &context.allocator);
+
+    // Create the work command buffer and fence for work unrelated to the swap chain.
+    VkCommandBufferAllocateInfo allocateInfo = {
+        .sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO,
+        .commandPool = context.commandPool,
+        .level = VK_COMMAND_BUFFER_LEVEL_PRIMARY,
+        .commandBufferCount = 1
+    };
+    VkFenceCreateInfo fenceCreateInfo { .sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO, };
```
Comment (bejado) [inline filament/src/driver/vulkan/VulkanDriverImpl.cpp]:
```suggestion
    VkFenceCreateInfo fenceCreateInfo { .sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO };
```

**Pull request merged**